conda_package_streaming/lazy_wheel.py

```diff
@@ -214,7 +272,12 @@ def _merge(
         self._left[left:right], self._right[left:right] = [start], [end]
 
     def _download(self, start: int, end: int) -> None:
-        """Download bytes from start to end inclusively."""
+        """
+        Download bytes from start to end inclusively. If the server does not support streaming for
+        this file, this does nothing as the entire file is already downloaded.
+        """
+        if not self._has_streaming_support:
```

Comment:
Can update self._right, self._left to encompass the whole file, then this if is not necessary and the below download code won't transfer anything

Reply:
Yes, but I want to note that `read(size: int = -1)` calls `_download` with incorrect arguments. If size is not set and read is immediately called after the constructor, read calls _download with start=10001 and stop=10000, assuming a file that is 10000 bytes large.

However, the uppermost valid value for stop is 9999, and start should be smaller than or equal than stop.

Despite these concerns, I removed the if-check in d1fd3605df0d2a258d0a58261fa40e2d28150278. left and right are actually already set by the constructor.